### PR TITLE
Document `repository` and `issue_tracker` properties

### PR DESCRIPTION
**Pull request by jonasfj**
As added in: https://github.com/dart-lang/pubspec_parse/pull/30

### CODE REVIEW

File: src/tools/pub/pubspec.md [hunk 1]
```diff
@@ -181,6 +189,24 @@ helps users understand where your package is coming from. If nothing else, you
 can always use the URL where you host the source code, such as
 [GitHub](https://github.com).
 
+### Repository
+
+The `repository` property should be a URL pointing to the repository for the
```
Comment (kwalrath) [inline src/tools/pub/pubspec.md]: Let's tighten this up a bit and refer to the homepage field. How about:

> This optional field should contain the URL for your package's source code — for example, `https://github.com/<user>/<repository>`. If you publish your package to pub.dartlang.org, then your package's page displays the repository URL. Also see the [homepage field](#homepage).

We should adjust the **Homepage** section as well to reflect how it interacts with `repository`.

@kevmoo, what should we say in the `homepage` section? Maybe change the last sentence from "If nothing else, you can always use the URL where you host the source code, such as GitHub." to something like this: "If nothing else, omit the [`repository` field](#repository) and set `homepage` to the URL where you host the source code."

File: src/tools/pub/pubspec.md [hunk 2]
```diff
@@ -181,6 +189,24 @@ helps users understand where your package is coming from. If nothing else, you
...
+package. This is often the URL for a github repository:
+`https://github.com/<user>/<repository>`.
+For [hosted packages](/tools/pub/dependencies#hosted-packages),
+this URL will be displayed on package's page.
+
+### Issue Tracker
```
Comment (kwalrath) [inline src/tools/pub/pubspec.md]: Tracker -> tracker
(we use sentence case for section heads)

File: src/tools/pub/pubspec.md [hunk 3]
```diff
@@ -181,6 +189,24 @@ helps users understand where your package is coming from. If nothing else, you
...
+
+The `issue_tracker` property should be a URL pointing to the issue tracker for
```
Comment (kwalrath) [inline src/tools/pub/pubspec.md]: How about:

Help your package's users find and file bugs in the right place by putting a URL in the `issue_tracker` field. The pub.dartlang.org site attempts to display a link to each package's issue tracker, using the value of this field. If `issue_tracker` is missing but [`repository`](#repository) is present and points to GitHub, then pub.dartlang.org uses the default issue tracker (`https://github.com/<user>/<repository>/issues`).

File: src/tools/pub/pubspec.md [hunk 4]
```diff
@@ -34,6 +34,14 @@ A pubspec can have the following fields:
 : Optional.
   [_Learn more._](#homepage)
 
+`repository`
+: Optional. URL pointing to the repository.
```
Comment (kwalrath) [inline src/tools/pub/pubspec.md]: repository -> the package's source code

(or something else that doesn't just repeat "repository". we could say only "Optional. Learn more." like some other entries, but I like your idea of giving _some_ info here.)

File: src/tools/pub/pubspec.md (hunk 1 above)
Comment (jonasfj) [reply src/tools/pub/pubspec.md]: Actually, I don't think we care if people provide both `homepage` and `repository`, or just one of them... either way is probably fine :)

So let's add our plea to both places:
> While this is technically optional *please do* provide either a `homepage`
or a `repository`. It helps users understand where your package is coming from.

File: src/tools/pub/pubspec.md [hunk 5]
```diff
@@ -176,10 +184,26 @@ public.
 This should be a URL pointing to the website for your package.
 For [hosted packages](/tools/pub/dependencies#hosted-packages),
 this URL is linked from the package's page.
-While this is technically optional *please do* provide one. It
-helps users understand where your package is coming from. If nothing else, you
-can always use the URL where you host the source code, such as
-[GitHub](https://github.com).
+While this is technically optional *please do* provide either a `homepage`
+or a `repository`. It helps users understand where your package is coming from.
+
```
Comment (kwalrath) [inline src/tools/pub/pubspec.md]: Both is OK, too, right? "either...or" makes it sound like you can't do both. How about:

While providing a `homepage` is optional, please provide it or `repository` (or both).
These fields help users understand...

File: src/tools/pub/pubspec.md [hunk 6]
```diff
@@ -176,10 +184,26 @@ public.
...
+### Repository
+
+The optional `repository` field should contain the URL for your package's source
+code repository — for example, `https://github.com/<user>/<repository>`.
+If you publish your package to pub.dartlang.org, then your package's page
+displays the repository URL.
+While this is technically optional *please do* provide either a `homepage`
```
Comment (kwalrath) [inline src/tools/pub/pubspec.md]: Same as above, except switch `homepage` and `repository`.

File: src/tools/pub/pubspec.md [hunk 7]
```diff
@@ -176,10 +184,26 @@ public.
...
+or a `repository`. It helps users understand where your package is coming from.
+
+### Issue tracker
+
+The optional `issue_tracker` field should contain a URL for the package's
+issue tracker, where existing bugs can be viewed and new bugs can be filed.
+The pub.dartlang.org site attempts to display a link to each package's issue
+tracker, using the value of this field. If `issue_tracker` is missing, but
+repository is present and points to GitHub, then pub.dartlang.org uses the
```
Comment (kwalrath) [inline src/tools/pub/pubspec.md]: , but repository -> but `repository`
(no comma, add code font)

File: src/tools/pub/pubspec.md (hunk 5 above)
Comment (jonasfj) [reply src/tools/pub/pubspec.md]: good point :)

**Comment by kwalrath** [comment]
LGTM. There's a tiny typo, but I'll fix that after committing. Thanks for the docs, @jonasfj!

**Pull request merged**